### PR TITLE
fix(input.redis_sentinel): fix sentinel and replica stats gathering

### LINKED ISSUE #12152: Replica and Sentinel stats are not gathered in redis_sentinel input plugin
### Relevant telegraf.conf

```toml
N/A
```

### Logs from Telegraf

```text
N/A
```

### System info

N/A

### Docker

_No response_

### Steps to reproduce

N/A

### Expected behavior

Replica and Sentinel stats are gathered in `redis_sentinel` input plugin

### Actual behavior

Replica and Sentinel stats are not gathered in `redis_sentinel` input plugin.

`gatherMasterStats` always returns empty `masterNames`.
image

Thus, `gatherReplicaStats` and `gatherSentinelStats` will never be called.
image

Introduced in this PR https://github.com/influxdata/telegraf/pull/10042 and I believe it wasn't intended.

@spideyfusion What do you think?

### Additional info

_No response_

### PR DESCRIPTION
**Pull request by spideyfusion**
# Required for all PRs

- [x] Updated associated README.md.
- [x] Wrote appropriate unit tests.
- [x] Pull request title or commits are in [conventional commit format](https://www.conventionalcommits.org/en/v1.0.0/#summary)

resolves #12152

---

I improved the integration test by making sure the plugin gets Initialized and talks to an actual Redis Sentinel instance.

### CODE REVIEW

File: plugins/inputs/redis_sentinel/redis_sentinel_test.go [hunk 1]
```diff
@@ -312,12 +346,35 @@ func TestRedisSentinelInfoAll(t *testing.T) {
...
 }
+
+func createRedisContainer() testutil.Container {
+	return testutil.Container{
+		Image:    "redis:7.0-alpine",
+		Name:     "telegraf-test-redis-sentinel-redis",
+		Networks: []string{networkName},
+		WaitingFor: wait.ForAll(
+			wait.ForExposedPort(),
+		),
```
Comment (powersj) [inline plugins/inputs/redis_sentinel/redis_sentinel_test.go:366]: Can you also have this wait for the log message "Ready to accept connections" as well? I like to see both a port + log. Similar to what you already do for the sentinels.

**Review by powersj — COMMENTED** [review]
Thank you for revamping this! One comment in line. Ideally you would continue to use the `testutil.Container` function, so I have on place to change things when testcontainers breaks backwards comparability again. However, I think this should be ok.

File: plugins/inputs/redis_sentinel/redis_sentinel_test.go (hunk 1 above)
Comment (spideyfusion) [reply plugins/inputs/redis_sentinel/redis_sentinel_test.go]: @powersj Done. I added the additional wait condition in https://github.com/influxdata/telegraf/pull/12229/commits/79c95e8dd4007780f01adfe1b9a01afb8f8e73d4.

**Review by srebhan — APPROVED** [review]
Looks good to me. Thanks for the fix @spideyfusion!

**Pull request merged**